### PR TITLE
Fix/unrandomized range filter tests

### PR DESCRIPTION
**Pull request by Bouncner**
Addresses #1321.

Remove randomness from range filter tests. One test I'd like to see is marked as a TODO for @Bensk1 : non-inclusive between predicates with the bounds exactly on the edges of the values ranges.

### CODE REVIEW

File: src/test/statistics/chunk_statistics/range_filter_test.cpp [hunk 1]
```diff
@@ -196,98 +162,119 @@ TYPED_TEST(RangeFilterTest, CanPruneOnBounds) {
...
 
-    // Additionally, test for further values in between the large random value ranges.
-    for (auto i = size_t{0}; i < 20; ++i) {
-      EXPECT_TRUE(
-          filter->can_prune(PredicateCondition::Equals, {get_random_number<TypeParam>(rng, middle_gap_distribution)}));
-    }
-  }
+  // For the default filter, the following value is prunable.
+  EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, 1));
+  // But malformed predicates are skipped intentially and are thus not prunable
```
Comment (mrzzzrm) [inline src/test/statistics/chunk_statistics/range_filter_test.cpp]: intentionally

File: src/test/statistics/chunk_statistics/range_filter_test.cpp [hunk 2]
```diff
@@ -196,98 +162,119 @@ TYPED_TEST(RangeFilterTest, CanPruneOnBounds) {
...
+  const pmr_vector<TypeParam> values{static_cast<TypeParam>(0.4 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.38 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.36 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.30 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.28 * std::numeric_limits<TypeParam>::lowest()),
+                                     static_cast<TypeParam>(0.36 * std::numeric_limits<TypeParam>::max()),
+                                     static_cast<TypeParam>(0.38 * std::numeric_limits<TypeParam>::max()),
+                                     static_cast<TypeParam>(0.4 * std::numeric_limits<TypeParam>::max())};
+
+  // A filter with 3 ranges, has two gaps: (i) 0.28*lowest-0.36*max and (ii) 0.36*lowest-0.30*lowest
```
Comment (mrzzzrm) [inline src/test/statistics/chunk_statistics/range_filter_test.cpp]: can you just add EXPECTs for this? I feel like that would already do 80% of the testing you're trying to achieve here.

File: src/test/statistics/chunk_statistics/range_filter_test.cpp [hunk 3]
```diff
@@ -196,98 +162,119 @@ TYPED_TEST(RangeFilterTest, CanPruneOnBounds) {
...
-
-  // We randomly create values between min_value(TypeParam) to 0.6*min_value(TypeParam) and
-  // 0.6*max(TypeParam) to max(TypeParam). Any value in between should be prunable.
-  distribution<TypeParam> lower_range_distribution(lower_bound_low_range, upper_bound_low_range);
-  distribution<TypeParam> high_range_distribution(lower_bound_high_range, upper_bound_high_range);
-  for (auto i = size_t{0}; i < 1'000; ++i) {
-    values.push_back(get_random_number<TypeParam>(rng, lower_range_distribution));
-    values.push_back(get_random_number<TypeParam>(rng, high_range_distribution));
-  }
+  const pmr_vector<TypeParam> values{static_cast<TypeParam>(0.4 * std::numeric_limits<TypeParam>::lowest()),
```
Comment (mrzzzrm) [inline src/test/statistics/chunk_statistics/range_filter_test.cpp]: Can you extract std::numeric_limits<TypeParam>::lowest() and std::numeric_limits<TypeParam>::max() into variables for readability

File: src/test/statistics/chunk_statistics/range_filter_test.cpp [hunk 4]
```diff
@@ -125,28 +70,69 @@ TYPED_TEST(RangeFilterTest, SingleRange) {
...
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, first_gap_min));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Between, first_gap_min, first_gap_max));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Equals, second_gap_min));
+    EXPECT_TRUE(filter->can_prune(PredicateCondition::Between, second_gap_min, second_gap_max));
+
+    EXPECT_FALSE(filter->can_prune(PredicateCondition::Between, third_gap_min, third_gap_max));
+  }
+  {
+    // starting with 4 ranges, all tested gaps should be covered
+    for (uint32_t range_count : {4, 5, 100, 1'000}) {
```
Comment (mrzzzrm) [inline src/test/statistics/chunk_statistics/range_filter_test.cpp]: just use auto here? I feel like sooner or later a compiler will start complaining about "assigning signed to non signed int literal"

**Comment by Bouncner** [comment]
I hope to have addressed all your comments.

**Comment by Bensk1** [comment]
I will let the seminar group know.

**Pull request merged**